src/utils/appLinking.js

```diff
@@ -0,0 +1,138 @@
+				: { uri: "linkedin://", url: "https://www.linkedin.com" }
+	),
+	slack: (inquiry, intent) => (
+		(intent === "channel" && (inquiry = (inquiry in slackInfo.channel) ? slackInfo.channel[inquiry] : inquiry))
+			? { uri: `slack://channel?team=${slackInfo.team}&id=${inquiry}` }
+			: { uri: `slack://open?team=${slackInfo.team}` }
+	)
+};
+
+const defIntentVal = (appName, intent) => {
+	if (!intent) {
+		const names = ["facebook", "instagram", "linkedin"];
+
+		intent = (names.includes(appName)) ? "profile" : "open";
+	}
+
+	return intent;
+};
+
+/**
+ * @description This function handles the action launching an app, like a mail app, from our app.
+ *              It uses uri schemes and universal links to do deep linking in both platforms.
+ *              **Requires two parameters and two optionals, action and warning message**.
+ *              With Slack, write the name of the channel or use slackInfo to see the channels available.
+ *
+ * @param {("email"|"facebook"
+ *         |"instagram"|"linkedin"
+ *         |"slack"|"phone"|"web")} appName  The app name to be opened.
+ * @param {("announcements"
+ *         |"fundraisingCommittee"
+ *         |"general"|"motorshpe"
+ *         |"projects"|"shpejr")}   inquiry  The value for the query search.
+ * @param { {intent: ("groups"
+ *            |"profile"|"company"
+ *            |"channel"|"open"),
+ *          warning: String}=}      config   (Optional) Intent for opening app and warning for alert.
+ *
+ * How to use
+ * ---
+ * 2 params: (appName, inquiry), pass the app name and what you are looking.
+ *
+ * 3 params: (appName, inquiry, { intent, warning }), pass the app name, what you are looking for,
+ *           action, and/or the alert message.
+ *
+ * @example
+ * // Slack: To open the chapter workspace
+ * openAppOrWebsite("slack", "team" | slackInfo.team, { warning: "Optional error message" });
+ * // Slack: To send the user directly to a channel
+ * openAppOrWebsite("slack", "general" | slackInfo.channel.general, { intent: "channel", warning: "Optional error message" });
+ *
+ * // LinkedIn (profile or company): To open a profile page
+ * openAppOrWebsite("linkedin", activeUser.linkedin | "companyProfile", { intent: "profile" | "company", warning: "Oops" })
+ *
+ * // Instagram (profile), Facebook (profile or group): To open a profile or group page
+ * openAppOrWebsite("appName", userProfile, { intent: "profile" | "group", warning: "Oops"})
+ *
+ * // Email, phone or website:
+ * openAppOrWebsite("mail" | "phone", activeUser.email | "14075551010", { warning: "Oops" })
+ * openAppOrWebsite("web", "https://www.example.com/", { warning: "Oops" })
+ */
+
+export function openAppOrWebsite(appName, inquiry, { intent, warning }) {
+	if (!inquiry) {
+		const name = appName[0].toUpperCase() + appName.slice(1);
+
+		Alert.alert(warning, { title: `Missing ${name} information.`, submit: { title: "Close" } });
+	}
```

Comment:
Eliminate this check, we assume the developer will call this function correctly and will use the documentation as a guide

Reply:
this check needs to be here. it is not there to take care of the coder but to catch the cases were a user hasnt enter their profile. it is the way to show that alert that says "the user hasnt enter their LinkedIn profile yet."